lib/services/plugins-service.ts

```diff
@@ -24,15 +25,21 @@ export class PluginsService implements IPluginsService {
 		private $options: IOptions,
 		private $logger: ILogger,
 		private $errors: IErrors,
-		private $projectFilesManager: IProjectFilesManager) { }
+		private $pluginVariablesService: IPluginVariablesService,
+		private $projectFilesManager: IProjectFilesManager,
+		private $injector: IInjector) { }
 
 	public add(plugin: string): IFuture<void> {
 		return (() => {
 			this.ensure().wait();
 			let dependencyData = this.$npm.cache(plugin, undefined, PluginsService.NPM_CONFIG).wait();
 			if(dependencyData.nativescript) {
+				let pluginData = this.convertToPluginData(dependencyData);
+				this.validate(pluginData).wait();
+
+				this.$pluginVariablesService.savePluginVariablesInProjectFile(pluginData).wait();
```

Comment:
In case this line of code or the one below it fails the project will be left in an unstable state. In order to avoid this a `try ... catch` statement with a logic for reversing changes inside the `catch` statement can be introduced